variables.tf

```diff
@@ -1,4 +1,26 @@
-variable "name" {
+variable "upstream_registry" {
   type        = string
-  description = "Describe what this variable is used for."
+  default     = "docker-hub"
+  description = "The upstream registry name"
+}
+
+variable "upstream_registry_url" {
+  type        = string
+  default     = "registry-1.docker.io"
+  description = "The upstream registry URL"
+}
+
+variable "aws_region" {
+  type        = string
+  default     = "eu-east-1"
+  description = "The AWS region to deploy the ECR pull through cache"
+}
+
+variable "credentials" {
+  type = map(string)
+  default = {
+    username    = "FILL-ME"
+    accessToken = "FILL-ME"
+  }
+  description = "Upstream registry credentials"
```

Comment:
I should avoid the usage of this variable. I would prefer to have these configuration statically used by the module itself. Accordingly to the ignore changes, after the first creation, if you change this value, the secret will be never updated.

Another consideration is to avoid the usage of private date here, so it is better if we remove the possibility to use them here.